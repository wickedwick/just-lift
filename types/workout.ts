export type Exercise = {
  id: string,
  name: string
  sets: number
  reps: number
  weight: number
  weightUnit: WeightUnit
  progressiveOverload: boolean
  overloadIncrement: number
}

export type Workout = {
  id: string
  exercises: Exercise[]
}

export type WorkoutPlan = {
  workouts: Workout[]
  daysPerWeek: number
  workoutIndex: number
  workoutInProgress: boolean
}

export type Log = {
  date: Date
  workoutId: string
  data: string
  exerciseName: string
  weight: number
  weightUnit: WeightUnit
}

export enum WeightUnit {
  Lbs,
  Kg
}
 