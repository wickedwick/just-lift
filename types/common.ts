import { Exercise, Log, WorkoutPlan } from './workout';
import { StyleProp, TextStyle, ViewStyle } from 'react-native';

export type RootStackParamList = {
  Root: undefined,
  NotFound: undefined
}

export type BottomTabParamList = {
  AutoTagger: undefined
  Celebrities: undefined
}

export type TabOneParamList = {
  BlankMenuScreen: undefined
  CreateWorkoutScreen: undefined
  SelectWorkoutScreen: undefined
  WorkoutScreen: undefined
  SettingsScreen: undefined
  ProfileScreen: undefined
  GuidedSelectionScreen: undefined
  WorkoutListScreen: undefined
}

export type TabTwoParamList = {
  TabOneScreen: undefined
  TabTwoScreen: undefined
}

export type WorkoutFormProps = {
  exercise?: Exercise
  onSubmit: (exercise: Exercise) => void
  onCancel: () => void
}

export type ExerciseCardProps = {
  exercise: Exercise
  onRemovePress: () => void
  onEditPress: () => void
}

export type ExerciseCounterProps = {
  exercise: Exercise
  setLogData: (log: Log) => void
  setWeight: (weight: number, exerciseName: string) => void
}

export type LogCardProps = {
  log: Log
}

export type ActionButtonProps = {
  onPress: () => void
  contained?: boolean
  icon?: string
  labelStyle?: StyleProp<TextStyle>
  style?: StyleProp<ViewStyle>
  text: string
}

export type RadioButtonProps = {
  label: string
  value: string
}

export type CheckboxProps = {
  label: string
  status: 'checked' | 'unchecked'
  onPress: () => void
}

export type WorkoutPlanContextProps = {
  workoutPlan: WorkoutPlan | null
  setWorkoutPlan: (workoutPlan: WorkoutPlan | null) => void
}
