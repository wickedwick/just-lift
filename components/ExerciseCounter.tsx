import React, { useEffect, useState } from 'react'
import RepInputGroup from './RepInputGroup'
import WeightEditor from './WeightEditor'
import { Card, Text } from 'react-native-paper'
import { ExerciseCounterProps } from '../types/common'
import { StyleSheet } from 'react-native'

const ExerciseCounter = (props: ExerciseCounterProps): JSX.Element => {
  const { exercise, setLogData, setWeight } = props
  const [counts, setCounts] = useState<number[]>([])

  useEffect(() => {
    const newCounts = []
    for(let i = 0; i < exercise.sets; i++) {
      newCounts.push(0)
    }

    setCounts(newCounts)
  }, [exercise])

  return (
    <Card style={styles.card}>
      <Card.Title title={exercise.name} />
      <Card.Content>
        <Text>{exercise.reps} reps</Text>

        <WeightEditor
          exercise={exercise}
          setWeight={setWeight}
        />
      </Card.Content>

      <Card.Content style={styles.flexContainer}>
        <RepInputGroup
          counts={counts}
          exercise={exercise}
          setCounts={setCounts}
          setLogData={setLogData}
        />
      </Card.Content>
    </Card>
  )
}

const styles = StyleSheet.create({
  card: {
    marginTop: 5,
    marginBottom: 5,
    borderTopLeftRadius: 0,
    borderTopRightRadius: 0,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
  },
  flexContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
})

export default ExerciseCounter
