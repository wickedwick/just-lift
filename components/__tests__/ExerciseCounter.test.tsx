import Adapter from 'enzyme-adapter-react-16'
import ExerciseCounter from '../ExerciseCounter'
import React from 'react'
import { Card } from 'react-native-paper'
import { configure, shallow } from 'enzyme'
import { Exercise, WeightUnit } from '../../types/workout'
import { ExerciseCounterProps } from '../../types/common'

configure({ adapter: new Adapter() })

const exercise: Exercise = {
  name: 'Bench Press',
  overloadIncrement: 5,
  progressiveOverload: false,
  reps: 5,
  sets: 3,
  weight: 200,
  weightUnit: WeightUnit.Lbs,
}

const props: ExerciseCounterProps = {
  exercise,
  setLogData: jest.fn(),
  setWeight: jest.fn()
}

describe('<ExerciseCounter />', () => {
  it('Renders with props', () => {
    const wrapper = shallow(<ExerciseCounter {...props} />)
    expect(wrapper.find(Card)).toHaveLength(1)
    expect(wrapper).toMatchSnapshot()
  })
})
