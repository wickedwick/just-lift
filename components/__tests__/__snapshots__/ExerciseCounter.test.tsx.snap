// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExerciseCounter /> Renders with props 1`] = `ShallowWrapper {}`;
